pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -75,7 +75,7 @@ func TestInt_Secrets(t *testing.T) {
 			objectassert.Secret(t, id).
 				HasName(id.Name()).
 				HasComment("a").
-				HasSecretType(sdk.SecretTypeOAuth2).
+				HasSecretType(string(sdk.SecretTypeOAuth2)).
```

Comment:
Question to the changes in this file: why enums were changed to strings? With asserts I guess it's a limitation of the generator, but what about secretDetails?

Reply:
It have changed in order to use `var AcceptableSecretTypes = map[SecretType]string` and pass secret types as sdk.SecretType type as it is done for the user_type CustomDiff `RecreateWhenUserTypeChangedExternally` 

But in case of secrets it is not really needed so it will be changed back